vnext/Microsoft.ReactNative/Utils/BatchingEventEmitter.cpp

```diff
@@ -90,13 +90,7 @@ void BatchingEventEmitter::EmitCoalescingJSEvent(
 
     isFirstEventInBatch = m_eventQueue.size() == 0;
 
-    auto endIter = std::remove_if(m_eventQueue.begin(), m_eventQueue.end(), [&](const auto &evt) noexcept {
-      return evt.eventEmitterName == newEvent.eventEmitterName && evt.emitterMethod == newEvent.emitterMethod &&
-          evt.eventName == newEvent.eventName && evt.coalescingKey == newEvent.coalescingKey;
-    });
-
-    m_eventQueue.erase(endIter, m_eventQueue.end());
-    m_eventQueue.push_back(std::move(newEvent));
+    AddOrCoalesceEvent(newEvent);
```

Comment:
Ooh, I think you also fixed an atomicity issue here. I had a bug here previously, where we could fire the batch before adding the event, after we just removed the previous version.